pkg/controller/operators/catalogtempate/operator.go

```diff
@@ -0,0 +1,241 @@
+package catalogtempate
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/distribution/distribution/reference"
+	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/workqueue"
+
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/api/client/clientset/versioned"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/api/client/informers/externalversions"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/lib/catalogsource"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/lib/operatorclient"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/lib/operatorlister"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/lib/queueinformer"
+)
+
+const (
+	StatusTypeTemplatesHaveResolved = "TemplatesHaveResolved"
+	StatusTypeResolvedImage         = "ResolvedImage"
+
+	ReasonUnableToResolve      = "UnableToResolve"
+	ReasonAllTemplatesResolved = "AllTemplatesResolved"
+)
+
+type Operator struct {
+	queueinformer.Operator
+	logger                        *logrus.Logger                  // common logger
+	namespace                     string                          // operator namespace
+	client                        versioned.Interface             // client used for OLM CRs
+	dynamicClient                 dynamic.Interface               // client used to dynamically discover resources
+	discoveryClient               *discovery.DiscoveryClient      // queries the API server to discover resources
+	lister                        operatorlister.OperatorLister   // union of versioned informer listers
+	catalogSourceTemplateQueueSet *queueinformer.ResourceQueueSet // work queues for a catalog source update
+	resyncPeriod                  func() time.Duration            // period of time between resync
+	ctx                           context.Context                 // context used for shutting down
```

Comment:
nit: From the [context package](https://pkg.go.dev/context#pkg-overview):

> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it. The Context should be the first parameter, typically named ctx

(although I know there are a few violations in OLM already)

Reply:
I realize storing the context is not ideal. This was necessary when I had the GVK code in place... the context was required in order to setup the watches, which would happen any time `syncCatalogSources` would fire. Now that the GVK code has been removed its not necessary, but will be necessary when adding the GVK code back someday.

Reply:
fixed